internal/lifecycle/lifecycle.go

```diff
@@ -221,6 +247,20 @@ func (l *Lifecycle) Stop(ctx context.Context) error {
 		return errors.New("called OnStop with nil context")
 	}
 
+	l.mu.Lock()
+	if l.state != started && l.state != incompleteStart {
+		l.mu.Unlock()
+		return errors.New("attempted to stop lifecycle when not running")
```

Comment:
this error message might make more sense if you printed the current state of `l.state` rather than just "not running" here since it could be either not running or starting or stopping.